### ISSUE TITLE
add GHA workflow - freetext > dropdown

### ISSUE THREAD

**Issue by rtc11** [ccmenu/ccmenu2#20]
We have 5K+ repositories in our org and I am presented with a drop down menu.

**Comment by erikdoe** [comment]
Yes, this is a known problem. Of course CCMenu could fetch all repositories, or the user could actively say "fetch more", but then a pop-up isn't a great UI for that. I wanted to use a search-box with drop-down but that UI element doesn't exist in Swift UI and it proved to be quite difficult to implement it. Will return to this sooner or later.

**Comment by rtc11** [comment]
Im not familiar with swift, but it looks like you save the state in json files? Where is the files located? I could problably add the repositories manually?

**Comment by erikdoe** [comment]
Yes, that's a possible workaround. That said, CCMenu stores its settings in the macOS database but you can export and import pipeline settings, and those are in JSON format. So, you can create a JSON file with your pipeline (or pipelines even) and then import that from the menu on the pipeline window in CCMenu. The format is relatively straightforward. Here's an example for CCMenu itself:

```
[
  {
    "feedName" : "",
    "feedType" : "github",
    "feedUrl" : "https://api.github.com/repos/ccmenu/ccmenu2/actions/workflows/build-and-test.yaml/runs",
    "name" : "ccmenu2 | Build and test"
  }
]
```

Note the outer angular brackets, the file must be an array of pipeline objects. The `feedName` field is always empty for GitHub pipelines.

**Comment by erikdoe** [comment]
This change has been made: The repository selector is now a combo box which allows you to enter any repository name. Will be release with CCMenu 27.0.

**Comment by erikdoe** [comment]
Please have a look whether this pre-release addresses your issue: https://github.com/ccmenu/ccmenu2/releases/tag/v27.0d1
